internal/pkg/cosign/ephemeral/signer.go

```diff
@@ -29,14 +29,14 @@ import (
 	"github.com/sigstore/sigstore/pkg/signature"
 )
 
-type keylessSigner struct {
+type ephemeralSigner struct {
 	signer signature.Signer
 }
 
-var _ icosign.Signer = keylessSigner{}
+var _ icosign.Signer = ephemeralSigner{}
 
 // Sign implements `Signer`
-func (ks keylessSigner) Sign(ctx context.Context, payload io.Reader) (oci.Signature, crypto.PublicKey, error) {
+func (ks ephemeralSigner) Sign(ctx context.Context, payload io.Reader) (oci.Signature, crypto.PublicKey, error) {
```

Comment:
You should probably use pointer receivers to avoid copies too

Reply:
copies of what? :p

Reply:
The struct containing the pointer 🙄